src/docfx/lib/log/Report.cs

```diff
@@ -54,27 +63,59 @@ public bool Write(Error error)
                 level = ErrorLevel.Error;
             }
 
-            if (_output != null)
+            if (ReachMaxErrors())
```

Comment:
>ReachMaxErrors() [](start = 16, length = 16)

`&& _maxExceeded == 0`? to avoid call `Interlocked.Exchange` after max exceeded.

Reply:
Performance wise it I guess there aren't that much difference. If you check `_maxExceeded == 0`, then `_maxExceeded` should be marked as `volatile`. `Interlocked.Exchange` is a single CPU instruction.

---
In reply to: [213230267](https://github.com/dotnet/docfx/pull/3298#discussion_r213230267) [](ancestors = 213230267)